src/test/java/htsjdk/samtools/SAMFileWriterFactoryTest.java

```diff
@@ -155,6 +155,10 @@ private void createSmallBamToOutputStream(final OutputStream outputStream,boolea
         final SAMFileWriterFactory factory = new SAMFileWriterFactory();
         factory.setCreateIndex(false);
         factory.setCreateMd5File(false);
+        final File wontBeUsed = new File("wontBeUsed.tmp");
+        final int maxRecsInRam = 271828;
+        factory.setMaxRecordsInRam(maxRecsInRam);
+        factory.setTempDirectory(wontBeUsed);
```

Comment:
Can you make a separate test case to test this functionality instead of modifying an existing one?